clustermesh/clustermesh.go

```diff
@@ -2082,6 +2073,107 @@ func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
 	return nil
 }
 
+// Helm-based clustermesh enable is only supported on Cilium
+// v1.14+ due to a lack of support in earlier versions for
+// autoconfigured certificates (tls.{crt,key}) for cluster
+// members when running in certgen (cronJob) PKI mode
+func (k *K8sClusterMesh) inClassicMode(r *release.Release) (bool, error) {
```

Comment:
nit: This would be better named `needsClassicMode` rather than `inClassicMode`, since we are definitely in Helm mode if we call this function.

Reply:
Thanks Andrew! I thought `ClassicMode` meant: "run the standard Clustermesh connect/disconnect as if helm upgrade mode is disabled".  If that's the case and the cilium rev is < 1.14.0 then we are `in classic mode'.
Am I wrong about that?

Reply:
Also do you have an idea on what's up with the tests failures? 
When I've first checked in this am, all tests were green and now 2 suites are failing consistently relating to a hosed tls secret key? Do you have a sense on why they started failing?